Share the same environment between the secret executable and the agent. (#2902)
DataDog/datadog-agent: docs/agent/secrets.md

- Have at least `exec` right for the owner.

Also:

> - The executable will share the same environment variables than agent.

Reviewer: ```suggestion
- The executable will share the same environment variables as the agent.
```

Reviewer: Okay, but what are the security implications, and why are we now doing this?

Author: The env is needed on windows. Many software (included python for example) need the environment to initialize things like random number generator.

Also this would mean sharing the environment of the agent user so the only sensitive information will most likely be the `API_KEY`. This information is available to the executable in any case: we exec the user software with the same user as the agent, which mean that executable can read the API token from the disk and query the agent configuration anyway. 

@mtremsal and @derekwbrown might have more input too.

Reviewer: Makes sense to me, we should just clarify

Cc @pratikgs-dd @jmassey-datadog

Author: Ping @trishankatdatadog @mtremsal. Can we merge this ?

- Be a valid Win32 application so the agent can execute it.

Also:

> - The executable will share any the same environment variables than agent.

Reviewer: ```suggestion
- The executable will share the same environment variables as the agent.
```

Reviewer: Okay, but what are the security implications, and why are we now doing this?

> - The executable will share any the same environment variables than agent.

Reviewer: ```suggestion
- The executable will share the same environment variables as the agent.
```
